### PR TITLE
Add Trace Detail page (Gantt Chart), upgrade Perses

### PR DESCRIPTION
**Pull request by andreasgerstmayr**
* upgrade Perses to v0.47.0-rc0
* support multiple routes (`/observe/traces/:traceId`)
* add `TraceDetailPage` including the `TracingGanttChart` from Perses

Resolves: https://issues.redhat.com/browse/OU-160

## Changes in Perses from 0.46 to 0.47.0-rc0:
Relevant changes:
```
[FEATURE] TracingGanttChart panel (#2047)
[FEATURE] TraceTable panel (#2061)
[FEATURE] TimeRange configurable #2055 (#2071)
[FEATURE] TimeSeries chart: add query-level settings for color override (#2046)
[ENHANCEMENT] Use DateTimeField in DateTimeRangePicker (#2093)
[ENHANCEMENT] update tracing data model (#2058)
[ENHANCEMENT] use relative bubble size in scatter plot (#2059)
[ENHANCEMENT] Increase empty dashboard message width (#2074)
[BUGFIX] Fix TraceQL query escaping (#2128)
[BUGFIX] Check content-type header before parsing error message as JSON (#2096)
[BUGFIX] Fix proxy endpoint that shouldn't be anonymous (#2062)
[BREAKINGCHANGE] Reword "Template Variable" as "Variable" everywhere (#2080)
[BREAKINGCHANGE] Rename various config to use the same wording (#2070)
[BREAKINGCHANGE] Refactor DateTimeRangePicker -> TimeRangeSelector + AbsoluteTimePicker -> DateTimeRangePicker (#2048)
```
Full list of changes: https://github.com/perses/perses/releases/tag/v0.47.0-rc0

## Screenshots
Link:
grafik

TraceDetail page:
grafik
grafik

### CODE REVIEW

File: web/src/TracingUI.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export default function TracingUI() {
+  return (
+    <Switch>
+      <Route path="/observe/traces" exact component={TracesPage} />
+      <Route
+        path="/observe/traces/:traceId"
+        exact
+        component={TraceDetailPage}
+      />
+      <Route component={TracesPage} />
```
Comment (zhuje) [inline web/src/TracingUI.tsx:15]: Is this needed if we already have route to the TracesPage  <Route path="/observe/traces" exact component={TracesPage} /> ?

**Comment by zhuje** [comment]
It looks like you added a new page, `TraceDetailPage`. Could you include a screenshot(s) of the new UI features?

File: web/webpack.config.ts [hunk 2]
```diff
@@ -70,6 +70,12 @@ const config: Configuration = {
     devMiddleware: {
       writeToDisk: true,
     },
+    client: {
+      overlay: {
+        warnings: false,
```
Comment (zhuje) [inline web/webpack.config.ts:75]: I'm hesitant to turn off the `warnings` and `errors`. I agree, it is annoying to see `WARNING in shared modules react` each time we render. But I'm concerned that other developers will be confused when they don't see the overlays rendered when there is a warning/error and it only appears in the console log. We could include this in the /hack files or as a note in the README.  WDYT?

File: web/src/TracingUI.tsx (hunk 1 above)
Comment (andreasgerstmayr) [reply web/src/TracingUI.tsx:15]: I kept it here as a catch-all route, if the above routes don't match.
But on second thought, you're right, probably we don't want a catch-all route.
I'll remove it.

File: web/webpack.config.ts (hunk 2 above)
Comment (andreasgerstmayr) [reply web/webpack.config.ts:75]: I got too annoyed seeing this warning on every code change (which auto-reloads the page) :grinning: . And the errors about unused imports etc., were also not helpful during development.

I'll revert it and just keep it locally.

**Comment by andreasgerstmayr** [comment]
> I believe the Perses `v0.47.0-rc0` includes [perses/perses PR#2058 [ENHANCEMENT] update tracing data model](https://github.com/perses/perses/pull/2058/files), which updates the tracing model to accommodate the Tempo vParquet4 block format. From my understanding, this PR updates the `TraceTable.tsx` to adapt to this new schema.

Yes. In a follow-up PR I'll update the Traces page to include the TraceTable from Perses. I didn't make the change in this PR to keep the PRs smaller and easier to review.

> Could you include a summary of the relevant changes from `v0.47.0-rc0` in the PR description?

Done.

**Pull request merged**